lang/py/avro/test/test_schema.py

```diff
@@ -634,13 +832,29 @@ def test_fixed_decimal_invalid_max_precision(self):
     def test_parse_invalid_symbol(self):
         """Disabling enumschema symbol validation should allow invalid symbols to pass."""
         test_schema_string = json.dumps({"type": "enum", "name": "AVRO2174", "symbols": ["white space"]})
-        with self.assertRaises(avro.errors.InvalidName, msg="When enum symbol validation is enabled, an invalid symbol should raise InvalidName."):
+
+        try:
             avro.schema.parse(test_schema_string, validate_enum_symbols=True)
+        except avro.errors.InvalidName:
+            pass
+        else:
+            self.fail("When enum symbol validation is enabled, an invalid symbol should raise InvalidName.")
```

Comment:
What is the motivation for switching from `assertRaises` to `try/except` in the tests?

Reply:
In this case, I use the `try/except` format to display a custom, friendlier message on failure.

Reply:
Doesn't the `msg` argument to `assertRaises` display that message?

I'm concerned about a couple of things here:
1. The happy path, where the test passes, actually has no assert statements in it. So it doesn't look like a test. Some test discovery tools I've worked with in the past will complain about that, or ignore the test altogether.
2. Since we don't want the test to fail in CI, there's a branch of this code that cannot be covered. (The `self.fail` part.) So that needs to be excluded from coverage.

Reply:
`unittest` documentation says this format should work: `assertRaises(exception, *, msg=None)`. The documentation also says `Changed in version 3.3: Added the msg keyword argument when used as a context manager.` Can you point me to an example of how to use `msg`?

Reply:
I think that's why I'm confused. The stuff on the left already has the `msg` keyword and as far as I was know it was working. (Was this perhaps something that didn't exist when this PR first came out?)

Reply:
Aah, I understand better now. I am unsure about the history of this piece of code - it has been a long time since I wrote it the first time. 🧐 

I have now replaced the `try/except` with `assertRaises` for the first part of the test and retained the second `try/except` to [check the exception was not raised](https://stackoverflow.com/a/4319870/1858466).

Reply:
The `# pragma: no coverage` is required for the second test because the exception will never happen in the codebase, resulting in <100% coverage.